### PR TITLE
Codeshare information in LFS Result

### PR DESCRIPTION
**Pull request by creduo**
This pr adds codeshare informations to segments in LFS Result

eg.
> "isCodeshare": true,
"operatingCarrier": "FM",
"operatingCarrierName": "SHANGHAI AIRLINES CO LTD",
"operatingFlightNumber": "829",

### CODE REVIEW

File: src/Services/Air/AirFormat.js [hunk 1]
```diff
@@ -45,6 +56,8 @@ function formatSegment(segment) {
     airline: segment.Carrier,
     flightNumber: segment.FlightNumber,
     uapi_segment_ref: segment.Key,
+    codeshare,
```
Comment (shmuga) [inline src/Services/Air/AirFormat.js]: I would name it `isCodeshare` to indicate that it's a boolean flag

Comment (creduo) [reply src/Services/Air/AirFormat.js]: changed to isCodeshare

File: src/Services/Air/AirFormat.js [hunk 2]
```diff
@@ -86,6 +86,17 @@ function getBaggageInfo(info) {
 }
 
 function formatSegment(segment) {
+  const isCodeshare = (Object.prototype.toString.call(segment['air:CodeshareInfo']) === '[object Object]');
+  let codeshareInfo = {};
```
Comment (dchertousov) [inline src/Services/Air/AirFormat.js:90]: Would prefer `const`

```javascript
const codeshareInfo = isCodeshare
  ? {
    operatingCarrier: segment['air:CodeshareInfo'].OperatingCarrier,
    operatingCarrierName: segment['air:CodeshareInfo']._,
    operatingFlightNumber: segment['air:CodeshareInfo'].OperatingFlightNumber,
  }
  : {};
```

Comment (dchertousov) [reply src/Services/Air/AirFormat.js:90]: With let it's always easy to mutate when it's not needed to.

File: src/Services/Air/AirFormat.js [hunk 3]
```diff
@@ -86,6 +86,17 @@ function getBaggageInfo(info) {
...
+
+  if (isCodeshare) {
+    codeshareInfo = {
+      operatingCarrier: segment['air:CodeshareInfo'].OperatingCarrier,
+      operatingCarrierName: segment['air:CodeshareInfo']._,
```
Comment (dchertousov) [inline src/Services/Air/AirFormat.js:95]: Will it be the full name of carrier or what?

File: src/Services/Air/AirFormat.js [hunk 4]
```diff
@@ -95,6 +106,8 @@ function formatSegment(segment) {
     airline: segment.Carrier,
     flightNumber: segment.FlightNumber,
     uapi_segment_ref: segment.Key,
+    isCodeshare,
+    ...codeshareInfo
```
Comment (dchertousov) [inline src/Services/Air/AirFormat.js:110]: I would either to keep codeshareInfo as structure here or to prepend codeshare keys with `codeshare`.
First option sounds more preferable to me.

File: src/Services/Air/AirFormat.js (hunk 3 above)
Comment (creduo) [reply src/Services/Air/AirFormat.js:95]: This will be full name of carrier -ish.

File: src/Services/Air/AirFormat.js (hunk 4 above)
Comment (creduo) [reply src/Services/Air/AirFormat.js:110]: Used Object.assign to add codeshare informations when isCodeshare is true

**Pull request closed without merging**